File: fp/haskell/quicksort.hs
quicksort :: (Ord a) => [a] -> [a]
quicksort [] = []
quicksort (x : xs) =
    let smallSorted = quicksort (filter (<= x) xs)
        biggerSorted = quicksort (filter (> x) xs)
     in smallSorted ++ [x] ++ biggerSorted

File: standard/miscellaneous/quicksort.py
def quicksort(a):
    if len(a) == 0:
        return []
    head = a[0]
    ss = quicksort([x for x in a[1:] if x <= head])
    bs = quicksort([x for x in a[1:] if x > head])
    return ss + [head] + bs


print(quicksort([1, 5, 4, 3, 2]))
